src/App.tsx

```diff
@@ -1,24 +1,143 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
 import { UserWarning } from './UserWarning';
+import { deleteTodo, getTodos, postTodo } from './api/todos';
+import { ErrorInfo } from './components/Errorinfo';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { ErrorType } from './types/ErrorType';
+import { FilterType } from './types/FilterType';
+import { NoIdTodo, Todo } from './types/Todo';
 
-const USER_ID = 0;
+const USER_ID = 12017;
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState<FilterType>(
+    FilterType.All,
+  );
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [errorMsg, setErrorMsg] = useState<ErrorType | null>(null);
+
+  const errorFound = (error: ErrorType) => {
+    setErrorMsg(error);
+    setTimeout(() => setErrorMsg(null), 3000);
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodosFromServer)
+      .catch(() => errorFound(ErrorType.TodosNotLoaded));
+  }, []);
+
+  const createNewTodo = (title: string): Promise<void> => {
+    const newTodo: NoIdTodo = {
+      userId: USER_ID,
+      title,
+      completed: false,
+    };
+
+    setTempTodo({
+      id: 0,
+      ...newTodo,
+    });
+
+    return postTodo(newTodo)
+      .then(
+        (response) => setTodosFromServer(
+          currentTodos => [...currentTodos, response],
+        ),
+      )
+      .catch(() => {
+        errorFound(ErrorType.NotAddable);
+
+        throw new Error(ErrorType.NotAddable);
+      })
+      .finally(() => setTempTodo(null));
+  };
+
+  const removeTodo = (todoToDelete: Todo) => {
```

Comment:
Loader should be visible also when todo is being deleted, i would recommend create a state for processing todo ids and add todo id every time you delete or add it so loader will be visible on that actions